dumpling/export/dump.go

```diff
@@ -1337,6 +1337,23 @@ func resolveAutoConsistency(d *Dumper) error {
 	default:
 		conf.Consistency = ConsistencyTypeNone
 	}
+
+	if conf.Consistency == ConsistencyTypeFlush {
+		// probe if upstream has enough privilege to FLUSH TABLE WITH READ LOCK
+		conn, err := d.dbHandle.Conn(d.tctx.Context)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+		err = FlushTableWithReadLock(d.tctx, conn)
```

Comment:
This means we will now have FTWRL twice?

Reply:
yes